src/string.cpp

```diff
@@ -302,6 +302,62 @@ bool StrValid(const char *str, const char *last)
 	return *str == '\0';
 }
 
+/**
+ * Trim the spaces from the begin of given string in place, i.e. the string buffer
+ * that is passed will be modified whenever spaces exist in the given string.
+ * When there are spaces at the begin, the whole string is moved forward.
+ * @param str The string to perform the in place left trimming on.
+ */
+static void StrLeftTrimInPlace(char *str)
+{
+	assert(str != nullptr);
+	char *first_non_space = str;
+	while (*first_non_space == ' ') first_non_space++;
+
+	if (first_non_space == str) return;
+
+	/* The source will reach '\0' first, but set the '\0' on the destination afterwards. */
+	char *dst = str;
+	for (char *src = first_non_space; *src != '\0'; dst++, src++) *dst = *src;
+	*dst = '\0';
+}
+
+/**
+ * Trim the spaces from the end of given string in place, i.e. the string buffer
+ * that is passed will be modified whenever spaces exist in the given string.
+ * When there are spaces at the end, the '\0' will be moved forward.
+ * @param str The string to perform the in place left trimming on.
+ */
+static void StrRightTrimInPlace(char *str)
+{
+	assert(str != nullptr);
+
+	char *end = str;
+	while (*end != '\0') end++;
+
+	char *last_non_space = end;
+	while (last_non_space > str && *last_non_space == ' ') last_non_space--;
+
+	if (last_non_space < str) {
+		*str = '\0';
+	} else {
+		last_non_space[1] = '\0';
+	}
```

Comment:
This if feels weird, but I cannot fully put my finger on it. `last_non_space` smaller than `str` means that there are trailing whitespaces not? So shouldn't this be inverted?
Also, I only expect a single if, as in, no else. As if there is no trailing whitespace, nothing changes, right?